test/test_autograd.py

```diff
@@ -2228,16 +2228,22 @@ def test_trapz(self):
                               True, f_args_variable, f_args_tensor)
 
     def test_cdist(self):
-        for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
-            f_args_variable = (torch.randn(S, S, requires_grad=True),
-                               torch.randn(S, S, requires_grad=True))
-
-            def f(a, b):
-                return torch.cdist(a, b, p)
-
-            f_args_tensor = deepcopy(unpack_variables(f_args_variable))
-            run_functional_checks(self, "test_cdist", "cdist", f,
-                                  True, f_args_variable, f_args_tensor)
+        def _test_cdist_for_size(sizes):
+            devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
+            for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
+                for device in devices:
+                    f_args_variable = (torch.randn(sizes, device=device, requires_grad=True),
+                                       torch.randn(sizes, device=device, requires_grad=True))
+
+                    def f(a, b):
+                        return torch.cdist(a, b, p)
+
+                    f_args_tensor = deepcopy(unpack_variables(f_args_variable))
+                    run_functional_checks(self, "test_cdist", "cdist", f,
+                                          True, f_args_variable, f_args_tensor)
+
+        _test_cdist_for_size((S, S))
+        _test_cdist_for_size((S, S, S))
```

Comment:
can you mix up the sizes?  There's a bunch of code above that grabs various sizes, but all of the sizes in your test are the same so you wouldn't notice a mistake there.